inductiva/api/methods.py

```diff
@@ -348,13 +360,92 @@ def invoke_api(method_name: str,
     return unpack_output(result_list, output_dir, return_type)
 
 
+def invoke_async_api(method_name: str, params, type_annotations: Dict[Any,
+                                                                      Type]):
```

Comment:
All the logic in this function is contained in `invoke_api`. Can we rewrite this to avoid repeating code?